test/testlib.py

```diff
@@ -905,8 +905,11 @@ def join_some(n):
                     if buffer:
                         sys.stdout.write(buffer.pop(pid))
                     pids.remove(pid)
-                if code:
-                    failures.append(code)
+                if code & 0xff:
+                    failed = 1
+                else:
+                    failed = (code >> 8) & 0xff
```

Comment:
this seems inconsistent with the way failures are returned https://github.com/cockpit-project/cockpit/pull/3071/files#diff-9b2c911d0ce5462bcaea62d679c32ec1R944